pkg/cmd/target.go

```diff
@@ -739,6 +769,137 @@ func gardenWrapper(targetReader TargetReader, targetWriter TargetWriter, configR
 	return nil
 }
 
+func serverWrapper(reader ConfigReader, serverName string, kubeconfigReader KubeconfigReader) {
+	config := reader.ReadConfig(pathGardenConfig)
+
+	for _, garden := range config.GardenClusters {
+		kc := garden.KubeConfig
+		svr := getServerValueFromKubeconfig(kc, kubeconfigReader)
+		equal, err := isServerEquals(serverName, svr)
+		if err != nil {
+			continue // skip error
+		}
+		if equal {
+			pgarden = garden.Name
+		}
+	}
+
+	if pgarden == "" {
+		fmt.Println("a garden could not be matched for the provided server address:", serverName)
+		os.Exit(2)
+	}
+
+}
+
+func isServerEquals(server1 string, server2 string) (bool, error) {
+	ips1, err := getIps(server1)
+	if err != nil {
+		return false, err
+	}
+
+	ips2, err := getIps(server2)
+	if err != nil {
+		return false, err
+	}
+
+	ipMatches := ipMatches(ips1, ips2)
+
+	port1, err := getPortForHost(server1)
+	if err != nil {
+		return false, err
+	}
+
+	port2, err := getPortForHost(server2)
+	if err != nil {
+		return false, err
+	}
+
+	return ipMatches && port1 == port2, nil
+}
+
+func getPortForHost(host string) (string, error) {
+	u, _ := url.ParseRequestURI(host)
+	if !hasPort(u.Host) {
+		if u.Scheme != "https" {
+			return "80", nil // assume http default port
+		}
+		return "443", nil
+	}
+	_, port, err := net.SplitHostPort(u.Host)
+	if err != nil {
+		return "", err
+	}
+	return port, nil
+}
+
+// returns true if at least one ip of ips1 is equal to an ip of ips2
+func ipMatches(ips1 []net.IP, ips2 []net.IP) bool {
+	for _, ip1 := range ips1 {
+		for _, ip2 := range ips2 {
+			if ip1.Equal(ip2) {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+func getIps(input string) ([]net.IP, error) {
+	ip := net.ParseIP(input)
+	if isIP := ip != nil; !isIP {
+		u, _ := url.ParseRequestURI(input)
+		host, err := getHostWithoutPort(u.Host)
+		if err != nil {
+			return nil, err
+		}
+
+		ips, err := net.LookupIP(host)
+		if err != nil {
+			return nil, err
+		}
+		return ips, nil
+	}
+	return []net.IP{ip}, nil
+}
+
+func hasPort(input string) bool {
+	// The port starts after the last colon.
+	i := strings.LastIndex(input, ":")
+	return i > 0
+}
+
+func getHostWithoutPort(input string) (string, error) {
+	if !hasPort(input) {
+		return input, nil
+	}
+
+	host, _, err := net.SplitHostPort(input)
+	if err != nil {
+		return "", err
+	}
+	return host, nil
+}
+
+func isValidURI(input string) bool {
+	_, err := url.ParseRequestURI(input)
+	return err == nil
+}
+
+func getServerValueFromKubeconfig(kubeconfigPath string, kubeconfigReader KubeconfigReader) string {
+	var kc Kubeconfig
+	if strings.Contains(kubeconfigPath, "~") {
+		kubeconfigPath = filepath.Clean(filepath.Join(HomeDir(), strings.Replace(kubeconfigPath, "~", "", 1)))
+	}
+	kubeconfig, err := kubeconfigReader.ReadKubeconfig(kubeconfigPath)
+	checkError(err)
+	err = yaml.Unmarshal(kubeconfig, &kc)
+	checkError(err)
+	if kc.Clusters[0].Cluster.Server == "" {
```

Comment:
I had another look at the code. Please take the server of the CurrentContext

Reply:
do you mean current-context in following screenshot? `kc.current-context`
this current-context is some kind of serviceaccount, not server name.

![image](https://user-images.githubusercontent.com/42594392/78156194-08717080-7471-11ea-8c78-116888bfa603.png)

Reply:
on second thought, you could create a ClientConfig like here
https://github.com/gardener/gardenctl/blob/master/pkg/cmd/miscellaneous.go#L108-L115
and get the host from the config 
`Host string`
https://godoc.org/k8s.io/client-go/rest#Config

Reply:
> do you mean current-context in following screenshot? `kc.CurrentContext `

yes, the current-context points to a context name. As said, you do not have to bother with it, just use the ClientConfig (see initialization here https://github.com/gardener/gardenctl/blob/master/pkg/cmd/miscellaneous.go#L108-L115) to read the Host.

Reply:
thanks, i've modified per your suggestion, also squash into one commit.